web_poet/serialization/api.py

```diff
@@ -189,5 +191,9 @@ def deserialize(cls: Type[InjectableT], data: SerializedData) -> InjectableT:
     for fn_or_cls, kwargs_spec in plan[:-1]:
         if fn_or_cls in externally_provided:
             continue
-        deps[fn_or_cls] = fn_or_cls(**kwargs_spec.kwargs(deps))
+        fn_or_cls = cast(Type[Injectable], fn_or_cls)
+        if fn_or_cls in DUMMY_MAP:
+            deps[fn_or_cls] = DUMMY_MAP[fn_or_cls]()
+            continue
+        deps[fn_or_cls] = fn_or_cls(**kwargs_spec.kwargs(deps))  # type: ignore
```

Comment:
mypy throws `web_poet/serialization/api.py:198: error: <nothing> not callable  [misc]` after I added `cast` above, which I added to prevent another complaint from mypy on `DUMMY_MAP[fn_or_cls]`.